### PR TITLE
fix: stop targeting all packages with --since if no changes

### PR DESCRIPTION
**Pull request by stevejcox**
## Current Behavior
If no changes are made to any package, `--since` will target all packages. 

## Expected Behavior
If no changes are made to any package, running `--since` should target no packages.

## To Reproduce
Branch off of main and attempt a build with `turbo run build --since=main`.

## Implementation Detail
This PR removes the loop which populates all packages if git diff turns up empty.

@jaredpalmer - let me know if this is currently intended behavior, it's unclear what the original intent of this loop was.

### CODE REVIEW

**Review by jaredpalmer — APPROVED** [review]

**Comment by stevejcox** [comment]
@jaredpalmer - anything I need to do to merge this? Looks like Kodiak is stuck waiting on checks that need manual approval to run?

**Comment by jaredpalmer** [comment]
nope, I had to press "approve and run again"

**Pull request merged**
